pkg/controller/cluster/cluster_controller.go

```diff
@@ -526,14 +502,14 @@ func (c *ClusterController) manageNodeGroups(nodeGroups []*clusteroperator.NodeG
 	return nil
 }
 
-func (c *ClusterController) manageNodeGroup(cluster *clusteroperator.Cluster, nodeGroup *clusteroperator.NodeGroup, clusterNodeGroup clusteroperator.ClusterNodeGroup, nodeType clusteroperator.NodeType, nodeGroupNamePrefix string) (*clusteroperator.NodeGroup, bool, error) {
-	if nodeGroup == nil {
-		return buildNewNodeGroup(cluster, clusterNodeGroup, nodeType, nodeGroupNamePrefix), false, nil
+func (c *ClusterController) manageMachineSet(cluster *clusteroperator.Cluster, machineSet *clusteroperator.MachineSet, clusterMachineSetSpec clusteroperator.MachineSetSpec, machineSetNamePrefix string) (*clusteroperator.MachineSet, bool, error) {
+	if machineSet == nil {
+		return buildNewMachineSet(cluster, clusterMachineSetSpec, machineSetNamePrefix), false, nil
 	}
 
-	if nodeGroup.Spec.Size != clusterNodeGroup.Size {
-		glog.V(2).Infof("Changing size of node group %s from %v to %v", nodeGroup.Name, nodeGroup.Spec.Size, clusterNodeGroup.Size)
-		return buildNewNodeGroup(cluster, clusterNodeGroup, nodeType, nodeGroupNamePrefix), true, nil
+	if machineSet.Spec.Size != clusterMachineSetSpec.Size {
```

Comment:
Presumably, there is more that we need to check here other than the size of the machine set. With the change to use a shared spec, it may be that we can get away with doing a `reflect.DeepEqual`. Of course, there may be changes to do not require a delete and re-create of the machine set and can instead do an update on the existing machine set.